extensions/resource-deployment/src/services/resourceTypeService.ts

```diff
@@ -245,6 +251,42 @@ export class ResourceTypeService implements IResourceTypeService {
 		return undefined;
 	}
 
+	/**
+	 * Get the ok button text based on the selected options
+	 */
+	private getOkButtonText(resourceType: ResourceType, selectedOptions: { option: string, value: string }[]): string | undefined {
+		if (resourceType.okButtonText) {
+			for (let i = 0; i < resourceType.okButtonText.length; i++) {
+				const buttonText = resourceType.okButtonText[i];
+				if (buttonText.when === undefined || buttonText.when.toString().toLowerCase() === 'true') {
+					return buttonText.value;
+				} else {
+					let expected = buttonText.when.replace(' ', '').split('&&').sort();
```

Comment:
I would recommend to make a function that compares the when string to selectedOptions instead of replicating the code.